lib/usd/utils/util.cpp

```diff
@@ -69,6 +71,49 @@ namespace
             {"nodePath", arc.GetTargetNode().GetPath().GetString()},
         };
     }
+
+    void replaceReferenceItems(const UsdPrim& oldPrim, 
+                               const SdfPath& newPath,
+                               const SdfReferencesProxy& referencesList,
+                               SdfListOpType op)
+    {
+        // set the itemVector based on the SdfListOpType
+        auto itemVector = referencesList.GetAppendedItems();
```

Comment:
Awkward that we assign to the itemVector unconditionally, then potentially squash it at lines 82-89.

Reply:
itemVector is of type SdfListProxy<SdfReferenceTypePolicy>and can't be created without initializing. 

`error C2512: 'pxrInternal_v0_20__pxrReserved__::SdfListProxy<pxrInternal_v0_20__pxrReserved__::SdfReferenceTypePolicy>': no appropriate default constructor available`

This was what I originally had in mind.
```
        // set the itemVector based on the SdfListOpType
        SdfReferencesProxy::ListProxy itemVector; 

        if (op == SdfListOpTypeAppended) {
            itemVector = referencesList.GetAppendedItems();
        } if (op == SdfListOpTypePrepended) {
            itemVector = referencesList.GetPrependedItems();
        } else if (op == SdfListOpTypeOrdered) {
            itemVector = referencesList.GetOrderedItems();
        } else if (op == SdfListOpTypeAdded) {
            itemVector = referencesList.GetAddedItems();
        } else if (op == SdfListOpTypeDeleted) {
            itemVector = referencesList.GetDeletedItems();
        }
```
I also went with auto to be on the safe side but the type here is `SdfListProxy<SdfReferenceTypePolicy> `  or a.k.a ( SdfReferencesProxy::ListProxy )

```
std::cout << "type = " << typeid(itemVector).name() << '\n';
outputs: SdfListProxy<SdfReferenceTypePolicy>
```

Also the name "itemVector" is a bit misleading which is the return type for GetAppendedItems, GetPrependedItems, GetOrderedItems, GetAddedItems, GetDeletedItems  in SdfListOp but what we have here is from SdfListEditorProxy. There is no Replace() in SdfListOp. I addressed this in 0618a8e to make it more clear.

See SdfListOp vs SdfListEditorProxy 

https://github.com/PixarAnimationStudios/USD/blob/ebac0a8b6703f4fa1c27115f1f013bb9819662f4/pxr/usd/sdf/listOp.h

https://github.com/PixarAnimationStudios/USD/blob/ebac0a8b6703f4fa1c27115f1f013bb9819662f4/pxr/usd/sdf/listEditorProxy.h

Reply:
O.K., you want ugly, I can give you ugly :)

SdfReferencesProxy::ListProxy itemVector = (op == SdfListOpTypeAppended) ? referencesList.GetAppendedItems() : (op == SdfListOpTypePrepended) ? referencesList.GetPrependedItems() : (op == SdfListOpTypeOrdered) ? referencesList.GetOrderedItems() : (op == SdfListOpTypeAdded) ? referencesList.GetAddedItems() : (op == SdfListOpTypeDeleted) ? referencesList.GetDeletedItems() : SdfReferencesProxy::ListProxy();

Reply:
Per our offline conversation, I will leave things as is to avoid making broader changed.